tests/internals/trigger_update_so/trigger_update_so_test.go

```diff
@@ -0,0 +1,426 @@
+	t.Log("--- test two triggers ---")
+	KubectlApplyWithTemplate(t, data, "scaledObjectTriggerTemplate", scaledObjectTriggerTemplate)
+
+	data.MetricValue = 1
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, 1, 180, 3),
+		"replica count should be %d after 3 minutes", 1)
+
+	KubectlApplyWithTemplate(t, data, "scaledObjectTwoTriggerTemplate", scaledObjectTwoTriggerTemplate)
+	// scale to max with k8s wl = second trigger
+	KubernetesScaleDeployment(t, kc, workloadDeploymentName, int64(maxReplicas), namespace)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, maxReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", maxReplicas)
+}
+
+// scales to max with kubernetes worload(second trigger), removes it and
+// scales to 3 replicas based on metric value (first trigger)
+func testRemoveTrigger(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- test remove trigger 2 -> 1 ---")
+	KubectlApplyWithTemplate(t, data, "scaledObjectTwoTriggerTemplate", scaledObjectTwoTriggerTemplate)
+	data.MetricValue = 5 // 3 replicas (midReplicas)
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+
+	KubernetesScaleDeployment(t, kc, workloadDeploymentName, int64(maxReplicas), namespace)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, maxReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", maxReplicas)
+
+	// update SO -> remove k8s wl == second trigger
+	KubectlApplyWithTemplate(t, data, "scaledObjectTriggerTemplate", scaledObjectTriggerTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, midReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", midReplicas)
+}
+
+// test 3 triggers scaling works including one cpu metric
+func testThreeTriggersWithCPU(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- test 3 triggers (with cpu) ---")
+
+	// update SO should scale up based on cpu
+	KubectlApplyWithTemplate(t, data, "scaledObjectThreeTriggerTemplate", scaledObjectThreeTriggerTemplate)
+
+	// scaling takes longer because of fetching of the cpu metrics (possibly increase iterations if needed)
+	data.MetricValue = 10
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, maxReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", maxReplicas)
+
+	data.MetricValue = 0
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+	// expect min replica count to be 1 since no other load is present and cpu is given
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, 1, 180, 3),
+		"replica count should be %d after 3 minutes", 1)
+}
```

Comment:
I prefer to avoid this case including CPU because is more unstable. Do we need this test case with 3 triggers? I mean, we have already changed the triggers section

Reply:
We need to have cpu or memory scaler in there as well, as these are special type of scalers, so we need to be sure that the combination of these with standard scalers

Reply:
let's try to use super obvious values, like 10% or something that will happen easily

Reply:
i removed part of test that relied on waiting for cpu metric as its main scaler to scale to because the test timed out waiting for it - hence it looks like it was unstable as it was. Now the cpu trigger is added, but replicas are scaled according to metricValue, to MaxReplicas, cpu is "just there"